packages/core/src/components/BaseInput/BaseInput.module.scss

```diff
@@ -0,0 +1,112 @@
+@import "../../styles/Typography";
+
+.wrapper {
+  width: 100%;
+  position: relative;
+  display: flex;
+  align-items: center;
+  flex-shrink: 0;
+  gap: var(--spacing-small);
+  padding-block: var(--spacing-xs);
+  padding-inline: var(--spacing-medium) var(--spacing-xs);
+
+  @include vibe-text(text1, normal);
+  @include smoothing-text;
+
+  outline: none;
+  border: 1px solid var(--ui-border-color);
+  border-radius: var(--border-radius-small);
+  color: var(--primary-text-color);
+  background-color: var(--secondary-background-color);
+  transition: border-color var(--motion-productive-medium) ease-in;
+
+  &.small {
+    height: 32px;
+    @include vibe-text(text2, normal);
+  }
+
+  &.medium {
+    height: 40px;
+  }
+
+  &.large {
+    height: 48px;
+    padding-block: var(--spacing-small);
+  }
+
+  &.rightThinnerPadding {
+    padding-inline-end: var(--spacing-medium);
+  }
+
+  &:hover {
+    border-color: var(--primary-text-color);
+  }
+
+  &:has(.input:active, .input:focus) {
+    border-color: var(--primary-color);
+  }
+
+  // TODO in next major remove withReadOnlyStyle, should only be determined by read-only existence on input
+  &:has(.input:read-only).withReadOnlyStyle {
+    background-color: var(--allgrey-background-color);
+    border: none;
+
+    .input {
+      background-color: var(--allgrey-background-color);
+    }
+  }
+
+  &:has(.input:disabled) {
+    cursor: not-allowed;
+    user-select: none;
+    border: none;
+    pointer-events: none;
+    background-color: var(--disabled-background-color);
+  }
+
+  &.success {
+    border-color: var(--positive-color);
+
+    &:hover {
+      border-color: var(--positive-color);
+    }
+
+    &:has(.input:active, .input:focus) {
+      border-color: var(--positive-color);
+    }
+  }
+
+  &:has(.input[aria-invalid="true"]) {
```

Comment:
Didn't we say we can't use `has` for now?

Reply:
added support for both browsers that do and do not support has